app/controllers/api/functions.php

```diff
@@ -800,14 +801,16 @@
         $function = Authorization::skip(fn() => $dbForProject->getDocument('functions', $functionId));
 
         if ($function->isEmpty()) {
-            throw new Exception('Function not found', 404);
+            throw new Exception('Function not found', 404, Exception::FUNCTION_NOT_FOUND);
         }
 
         if (!empty($cursor)) {
             $cursorExecution = $dbForProject->getDocument('executions', $cursor);
 
             if ($cursorExecution->isEmpty()) {
-                throw new Exception("Execution '{$cursor}' for the 'cursor' value not found.", 400);
+                // TODO: Shouldn't this be a 404 error ? 
+                // Or do we define a new error type GENERAL_CURSOR_NOT_FOUND for these types ?
```

Comment:
@eldadfux  need your ideas on this

Reply:
GENERAL_ARGUMENT_INVALID ....

Reply:
I chose to go with `GENERAL_CURSOR_NOT_FOUND` 
since `GENERAL_ARGUMENT_INVALID` can be quite misleading / confusing. 

Argument invalid is used to refer to issues with Utopia arguments. 
Whereas a cursor not found can be because the cursor ID was deleted or the order of documents was changed.